keps/sig-api-machinery/20190324-declatative-field-manager.md

```diff
@@ -0,0 +1,176 @@
+---
+title: declarative-field-manager
+authors:
+  - "@kwiesmueller"
+owning-sig: sig-api-machinery
+reviewers:
+  - "@apelisse"
+  - "@lavalamp"
+approvers:
+  - "@lavalamp"
+creation-date: 2019-03-24
+last-updated: yyyy-mm-dd
+status: provisional
+---
+
+# declarative-field-manager
+
+## Summary
+
+Currently the identifier of the current actor (fieldManager) is set through an request option or the request user-agent,
+this KEP aims to provide a declarative way of setting the current fieldManager through a field in the object itself.
+
+## Motivation
+
+It is a common practice to use Kubernetes manifests as stored configuration of an application. Either to just introduce version control to the cluster's state, or to automate the process of updating objects.
+
+With the addition of Server Side Apply it could become necessary to change the way those manifests get applied, as the most benefit from this new feature requires every actor to define it's fieldManager. Doing this, currently is only possible through api options, the user-agent or as a kubectl option.
+
+To keep the way of interacting with manifests as easy as possible, this KEP suggests to allow defining the fieldManager as part of the manifest itself. Therefore no further action is required when applying a set of manifests and users can easily benefit from Server Side Apply without having to change their apply workflow.
+
+This would also assist existing solutions already interacting with manifest files, as the fieldManager information gets provided in a way they already know to handle which might improve compatibility.
+
+### Goals
+
+- Allow users to set their fieldManager information declaratively in their manifests
+
+### Non-Goals
+
+- Replace the existing ways to set the fieldManager information
+
+## Proposal
```

Comment:
Thanks, this section is a clear summary now.

I am in favor of this. I think I'd like @liggitt, as the most active (?) API reviewer to read this section and LGTM, too. But I think he has a pending comment below that hasn't been answered.

Reply:
This feels very specific to an anticipated user behavior and need, but do we have real evidence we need this *now*?  This is kind of... interesting, feels very specific to a fairly new feature, and while I could see us needing this I'd almost prefer to wait for this to become more of a problem before we move forward with it.